lib/fastly_nsq/manager.rb

```diff
@@ -1,104 +1,80 @@
 # frozen_string_literal: true
 
-require 'set'
-
 class FastlyNsq::Manager
-  attr_reader :listeners
+  DEADLINE = 30
+  DEFAULT_POOL_SIZE = 5
+
+  attr_reader :done, :pool, :logger
 
-  def initialize(options = {})
-    @options = options
-    @done = false
-    @listeners = Set.new
-    @plock = Mutex.new
+  def initialize(logger: FastlyNsq.logger, **pool_options)
+    @done      = false
+    @logger    = logger
+    @pool      = FastlyNsq::PriorityThreadPool.new(
+      { fallback_policy: :caller_runs, max_threads: DEFAULT_POOL_SIZE }.merge(pool_options),
```

Comment:
are we comfortable with `fallback_policy` being overridden by pool_options?

Reply:
You could do some fancy things with the thread pool so I wanted to leave the option open for other consumers.  Definitely advanced territory though.